qiskit/transpiler/coupling.py

```diff
@@ -159,19 +160,19 @@ def neighbors(self, physical_qubit):
     @property
     def distance_matrix(self):
         """Return the distance matrix for the coupling map."""
-        if self._dist_matrix is None:
-            self._compute_distance_matrix()
-        return self._dist_matrix
+        self.compute_distance_matrix()
+        return self.dist_matrix
 
-    def _compute_distance_matrix(self):
+    def compute_distance_matrix(self):
         """Compute the full distance matrix on pairs of nodes.
 
-        The distance map self._dist_matrix is computed from the graph using
+        The distance map self.dist_matrix is computed from the graph using
         all_pairs_shortest_path_length.
         """
         if not self.is_connected():
             raise CouplingError("coupling graph not connected")
-        self._dist_matrix = rx.digraph_distance_matrix(self.graph, as_undirected=True)
+        if self.dist_matrix is None:
+            self.dist_matrix = rx.digraph_distance_matrix(self.graph, as_undirected=True)
```

Comment:
In the interests of speed, the `self.is_connected` test should probably be inside the `if self.dist_matrix is None` test; `is None` tests are lightning fast and if the matrix is defined, then we don't need to re-test connectivity.  This will have more of an impact in the new access style, since `compute_distance_matrix` gets called more now.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/15e90e2d01c51713e54c5a0e6df8cc99645c52a9